### ISSUE TITLE
[TSQL] Parsing errors with cursors

### ISSUE THREAD

**Issue by oaklsb** [sqlfluff/sqlfluff#7098]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Trying to lint a file containing cursors fails with parsing error.

### Expected Behaviour

Should lint file

### Observed Behaviour

The file cannot be parsed. The below error message appears

```bash
$ sqlfluff lint cursor.sql 
== [cursor.sql] FAIL                                                                                                                                                    
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'declare
                       | @cursor cursor\nset @cursor = cur...'
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

Create file with cursor
```tsql
-- cursor.sql

declare @cursor cursor
set @cursor = cursor fast_forward
for
select
    my_column
from my_table

declare @MyColumn int

open @cursor
fetch next from @cursor into @MyColumn

while (@@FETCH_STATUS = 0)
    begin
        print @MyColumn
        fetch next from @cursor into @MyColumn
    end
close @cursor
deallocate @cursor
```
Run `sqlfluff lint cursor.sql`

### Dialect

TSQL

### Version

sqlfluff, version 3.4.2

### Configuration

```ini
; .sqlfluff

[sqlfluff]
dialect = tsql
templater = raw
exclude_rules = capitalisation.identifiers

max_line_length = 0
processes = -1

[sqlfluff:indentation]
allow_implicit_indents = True

[sqlfluff:rules:aliasing.length]
min_alias_length = 3

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower
[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower
[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower
[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = lower

[sqlfluff:rules:convention.not_equal]
preferred_not_equal_style = c_style

[sqlfluff:rules:layout.newlines]
maximum_empty_lines_between_statements = 1
maximum_empty_lines_inside_statements = 0
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
